lib/config/guiconfiguration.go

```diff
@@ -166,13 +166,15 @@ func (c *GUIConfiguration) prepare() error {
 		c.APIKey = rand.String(32)
 	}
 
-	if c.WebauthnUserId == "" {
-		newUserId := make([]byte, 64)
+	if len(c.WebauthnUserId) == 0 {
+		// Spec recommends 64 random bytes, we use 32 and hex-encoding inflates to 64
```

Comment:
The amount of entropy is still half of what is recommended, because hex-encoding only "inflates" the number of octets stored, but simultaneously reduces the number of possible code points (`[0-9a-f]`).  So we can't really call it "64 random bytes".  For the same amount of entropy, you'd need hex-encode 64 bytes.

Reply:
The comment doesn't make any claim on entropy?

I also don't find the randomness recommendation in the spec, just this:

> The [user handle](https://www.w3.org/TR/webauthn-2/#user-handle) of the user account entity. A [user handle](https://www.w3.org/TR/webauthn-2/#user-handle) is an opaque [byte sequence](https://infra.spec.whatwg.org/#byte-sequence) with a maximum size of 64 bytes, and is not meant to be displayed to the user.

Neither do I see a need for this to be high entropy - it's not a secret.

Reply:
Alright, I assumed the requirement was for actual entropy / "randomness".  In that case it's fine.

Reply:
Yeah, 32 bytes (256 bits) is more than enough for the value to be probabilistically unique within the application which is all it needs to be. The comment may be a bit misleading, I'll take another pass at it.

Reply:
Comment addressed in a2b4c30e827981a08ce000c96b6eff408478ed0d.